src/main/java/domain/Lotto.java

```diff
@@ -1,59 +1,56 @@
 package domain;
 
-import static domain.properties.LottoProperties.COUNT_OF_NUMBERS;
-import static domain.properties.LottoProperties.MAX_NUMBER;
-import static domain.properties.LottoProperties.MIN_NUMBER;
+import static domain.properties.LottoProperties.COUNT_OF_LOTTO_NUMBERS;
 
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 public class Lotto {
-    private final List<Integer> numbers;
+    private final List<LottoNumber> numbers;
 
-    private Lotto(List<Integer> numbers) {
+    private Lotto(List<LottoNumber> numbers) {
         validate(numbers);
         this.numbers = numbers.stream().sorted().toList();
     }
 
-    public static Lotto of(final List<Integer> numbers) {
+    public static Lotto of(final List<LottoNumber> numbers) {
         return new Lotto(numbers);
     }
 
-    private void validate(List<Integer> numbers) {
-        if (numbers == null || numbers.size() != COUNT_OF_NUMBERS) {
-            throw new IllegalArgumentException(String.format("로또 번호는 %d개여야 합니다.", COUNT_OF_NUMBERS));
+    private void validate(List<LottoNumber> numbers) {
+        if (numbers == null || numbers.size() != COUNT_OF_LOTTO_NUMBERS) {
+            throw new IllegalArgumentException(String.format("로또 번호는 %d개여야 합니다.", COUNT_OF_LOTTO_NUMBERS));
         }
 
-        Set<Integer> distinctNumbers = new HashSet<>(numbers);
-        if (distinctNumbers.size() != COUNT_OF_NUMBERS) {
+        if (containsDuplicate(numbers)) {
             throw new IllegalArgumentException("로또 번호는 중복될 수 없습니다.");
         }
+    }
 
-        for (Integer number : distinctNumbers) {
-            validateRange(number);
-        }
+    private boolean containsDuplicate(List<LottoNumber> lottoNumbers) {
+        List<LottoNumber> distinct = lottoNumbers.stream().distinct().toList();
+        return lottoNumbers.size() != distinct.size();
     }
 
-    private static void validateRange(final Integer number) {
-        if (number < MIN_NUMBER || number > MAX_NUMBER) {
-            throw new IllegalArgumentException(String.format("로또 번호는 %d부터 %d사이여야 합니다.", MIN_NUMBER, MAX_NUMBER));
-        }
+    public int match(Lotto compared) {
+        return (int) numbers.stream()
+                .filter(number -> containsSameValue(compared.numbers, number))
+                .count();
     }
 
-    public boolean contains(final int number) {
-        return numbers.contains(number);
+    public boolean contains(final LottoNumber number) {
+        return containsSameValue(this.numbers, number);
     }
 
-    public int match(Lotto compared) {
-        Set<Integer> distinctNumbers = new HashSet<>();
-        distinctNumbers.addAll(compared.numbers);
-        distinctNumbers.addAll(numbers);
-        return ((2 * COUNT_OF_NUMBERS) - distinctNumbers.size());
+    private boolean containsSameValue(List<LottoNumber> numbers, LottoNumber number) {
+        return numbers.stream()
+                .anyMatch(element -> element.isSameValue(number));
```

Comment:
equals 도 재정의 해주셨으니 사용해보는것은 어떻게 생각하시나요?

```suggestion
                .anyMatch(element -> element.equals(number));
```

++) element 보다는 더 적절한 변수명을 사용했으면 좋겠어요!